apps/metrics/__init__.py

```diff
@@ -0,0 +1,107 @@
+class MetricType:
+    UNKNOWN = 'unknown'
+    TIME = 'time'
+    SIZE = 'size'
+    COUNT = 'count'
+
+
+class MetricCollection(object):
+    def __init__(self, metrics=None):
+        if metrics is None:
+            metrics = []
+        self.metrics = metrics
+
+    def add(self, metric):
+        self.metrics.append(metric)
+
+    def __iter__(self):
+        return iter(self.metrics)
+
+    def sort(self, *args, **kwargs):
+        return MetricCollection(sorted(self.metrics, *args, **kwargs))
+
+    def filter(self, function):
+        return MetricCollection(filter(function, self.metrics))
+
+    def __add__(self, other):
+        new_collection = MetricCollection()
+        new_collection += self
+        new_collection += other
+        return new_collection
+
+    def __iadd__(self, other):
+        self.metrics.extend(other.metrics)
+
+    def serialize(self):
```

Comment:
We should be able to serialize each of those objects (MetricCollection and Metric) to JSON. I've run into this problem several times already... If we derive from list it should be automatic.